powerauth-restful-security-spring/src/main/java/io/getlime/security/powerauth/rest/api/spring/annotation/PowerAuthAnnotationInterceptor.java

```diff
@@ -128,8 +117,9 @@ public boolean preHandle(HttpServletRequest request, HttpServletResponse respons
 
             // Resolve @PowerAuthEncryption annotation
             if (powerAuthEncryptionAnnotation != null) {
+                Class<?> requestType = resolveGenericParameterTypeForEcies(handlerMethod);
```

Comment:
This method makes no sense to me - we are looking for the method parameter with `@EncryptedRequestBody` here, why do we iterate over all parameters, instead of doing the mapping at the place we know the type (argument resolver)?

Reply:
Iterating over parameters is done due to initialization of `PowerAuthEciesEncryption<T>` which is created before reaching argument resolver.